be/src/vec/exec/join/vhash_join_node.cpp

```diff
@@ -184,19 +179,21 @@ struct ProcessHashTableProbe {
 
         KeyGetter key_getter(_probe_raw_ptrs, _join_node->_probe_key_sz, nullptr);
 
-        std::vector<uint32_t> items_counts(_probe_rows);
+        uint32_t items_counts[_probe_rows];
         auto& mcol = mutable_block.mutable_columns();
         int current_offset = 0;
+        //build_block_index size may be largger than _batch_size, 100 is safety.
+        int build_block_row_num[_batch_size + 100];
```

Comment:
it's dangerous to have a pre safety set for noknown data, may should use vector.